astropath/scripts/use_catalogs.py

```diff
@@ -0,0 +1,156 @@
+    parser.add_argument("--ltype", type=str, default='ellipse', help="Localization type [ellipse] FUTURE: wcs, healpix")
+    parser.add_argument("-U", "--PU", type=float, default=0., help="Prior on unseen galaxies")
+    parser.add_argument("-s", "--survey", type=str, default='Pan-STARRS',
+                        help="Public survey to use for the analysis ['Pan-STARRS', 'DECaL']")
+    parser.add_argument("--ssize", type=float, default=5., help='Size of the survey in arcmin')
+    parser.add_argument("--debug", default=False, action="store_true", help="debug?")
+    parser.add_argument("-o", "--outfile", type=str, help="Name of the output file.  Should end in .csv")
+
+    if options is None:
+        pargs = parser.parse_args()
+    else:
+        pargs = parser.parse_args(options)
+    return pargs
+
+
+def main(pargs):
+    """ Run
+    """
+    import numpy as np
+
+    import seaborn as sns
+    from matplotlib import pyplot as plt
+
+    from astropy import units
+
+    from frb.surveys import survey_utils
+
+    from astropath import path
+    from astropath.scripts.utils import coord_arg_to_coord
+    from astropath.utils import radec_to_coord
+
+    scale = 0.5
+
+    if pargs.ltype == 'ellipse':
+        a, b, pa = [float(ip) for ip in pargs.lparam.split(',')]
+        eellipse = {'a': a, 'b': b, 'theta': pa}
+
+    # Load up the survey
+    coord = radec_to_coord(coord_arg_to_coord(pargs.coord))
+    survey = survey_utils.load_survey_by_name(
+        pargs.survey, coord, pargs.ssize*units.arcmin)
+
+    # Survey specific queries
+    if pargs.survey == 'Pan-STARRS':
+        query_fields = ['rPSFLikelihood']
+    elif pargs.survey == 'DECaL':
+        query_fields = ['shapedev_r', 'shapeexp_r']
+
+    # Grab the catalo
+    catalog = survey.get_catalog(query_fields=query_fields)
+
+    if len(catalog) == 0:
+        print(f"No objects in the catalog of your survey within {pargs.ssize} arcmin")
+        return
+
+    # Clean up the catalog
+    if pargs.survey == 'Pan-STARRS':
+        cut_size = catalog['rKronRad'] > 0.
+        cut_mag = catalog['Pan-STARRS_r'] > 14. # Reconsider this
+        cut_point = np.log10(np.abs(catalog['rPSFLikelihood'])) < (-2)
+        keep = cut_size & cut_mag & cut_point
+        # Half-light radius
+        mag_key = 'Pan-STARRS_r'
+        catalog['ang_size'] = catalog['rKronRad'].copy() 
+    elif pargs.survey == 'DECaL':
+        mag_key = 'DECaL_r'
+        # Cuts
+        cut_mag = (catalog[mag_key] > 14.) & np.isfinite(catalog[mag_key]) 
+        cut_star = catalog['gaia_pointsource'] == 0
+        keep = cut_mag & cut_star
+        # Half-light radius
+        catalog['ang_size'] = np.maximum(catalog['shapedev_r'], catalog['shapeexp_r'])
+        zero = catalog['ang_size'] == 0.
+        catalog['ang_size'][zero] = 1. # KLUDGE!!
```

Comment:
Probably should warn users if one of these galaxies gets a high PATH probability. I'd add a flag to the catalog that keeps track of this and warns later if P_Ox is high for one of these sources.

Reply:
good suggestion. I tried doing something like this but it was painful.

Am tripping a warning instead..